### ISSUE TITLE
Error handling in Apis

### ISSUE THREAD

**Issue by Archiit19** [gusaul/go-dynamock#27]
Instead of printing API errors in case of empty Inputs, we should allow users to set error in the APIs that would give flexibility and consistency with the AWS API responses.

**Comment by stan-irl** [comment]
yeah id like this as well. right now its not possible to replicate `TransactionCanceledException` or `ConditionalCheckFailedException` which makes it impossible to test these codepaths.

I imagine most people who find this repo are looking for ways to achieve 100% code coverage (well at least thats how I found it)
